src/main/java/cz/mzk/fedora/documentcorrection/RestClient.java

```diff
@@ -0,0 +1,117 @@
+package cz.mzk.fedora.documentcorrection;
+
+import org.apache.tomcat.util.codec.binary.Base64;
+import org.springframework.http.*;
+import org.springframework.web.client.RestTemplate;
+import org.w3c.dom.*;
+import org.xml.sax.InputSource;
+import org.xml.sax.SAXException;
+
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+import javax.xml.transform.Transformer;
+import javax.xml.transform.TransformerException;
+import javax.xml.transform.TransformerFactory;
+import javax.xml.transform.dom.DOMSource;
+import javax.xml.transform.stream.StreamResult;
+import javax.xml.xpath.*;
+import java.io.IOException;
+import java.io.StringReader;
+import java.io.StringWriter;
+import java.util.Objects;
+
+public class RestClient {
+
+    private final String fedoraHost;
+    private final HttpEntity<String> httpEntity;
+    private final RestTemplate restTemplate;
+    private final DocumentBuilder xmlParser;
+    private final Transformer xmlTransformer;
+    private final XPathExpression xmlPathExp;
+
+    public RestClient(String fh, String fu, String fp)
+            throws ParserConfigurationException, XPathExpressionException, TransformerException {
+        fedoraHost = fh;
+        restTemplate = new RestTemplate();
+        HttpHeaders httpHeaders = createHttpHeaders(fu, fp);
+        httpEntity = new HttpEntity<>(httpHeaders);
+
+        DocumentBuilderFactory factory = DocumentBuilderFactory.newDefaultInstance();
+        factory.setNamespaceAware(true);
+        xmlParser = factory.newDocumentBuilder();
+
+        XPathFactory xFactory = XPathFactory.newInstance();
+        XPath xmlPath = xFactory.newXPath();
+        xmlPathExp = xmlPath.compile("//*[local-name() = 'isMemberOfCollection']");
```

Comment:
Pro parsování XML elementů bych udělal další pomocný servis. Těch XPathExpressions bude víc. Ale zatim at to zůstane takto.